src/ngraph/op/strided_slice.cpp

```diff
@@ -227,3 +232,154 @@ void op::v1::StridedSlice::generate_adjoints(autodiff::Adjoints& /* adjoints */,
 {
     throw ngraph_error("generate_adjoints not implemented for StridedSlice");
 }
+
+namespace
+{
+    template <element::Type_t ET>
+    inline bool evaluate(const HostTensorPtr& in, const SlicePlan& sp, const HostTensorPtr& out)
+
+    {
+        auto in_shape = in->get_shape();
+        out->set_shape(sp.reshape_out_shape);
+        runtime::reference::strided_slice(
+            in->get_data_ptr<ET>(), out->get_data_ptr<ET>(), in_shape, sp);
+        return true;
+    }
+
+    bool evaluate_strided_slice(const HostTensorPtr& in,
+                                const HostTensorPtr& begin,
+                                const HostTensorPtr& end,
+                                const HostTensorPtr& stride,
+                                const AxisSet& begin_mask,
+                                const AxisSet& end_mask,
+                                const AxisSet& new_axis_mask,
+                                const AxisSet& shrink_axis_mask,
+                                const AxisSet& ellipsis_mask,
+                                const HostTensorPtr& out)
+    {
+        bool rc = true;
+
+        std::vector<int64_t> begin_const = read_vector<int64_t>(begin);
+        std::vector<int64_t> end_const = read_vector<int64_t>(end);
+        std::vector<int64_t> stride_const = read_vector<int64_t>(stride);
+        SlicePlan slice_plan = make_slice_plan(in->get_shape(),
+                                               begin_const,
+                                               end_const,
+                                               stride_const,
+                                               begin_mask,
+                                               end_mask,
+                                               new_axis_mask,
+                                               shrink_axis_mask,
+                                               ellipsis_mask);
+        switch (in->get_element_type())
+        {
+            TYPE_CASE(i8)(in, slice_plan, out);
+            break;
+            TYPE_CASE(i16)(in, slice_plan, out);
+            break;
+            TYPE_CASE(i32)(in, slice_plan, out);
+            break;
+            TYPE_CASE(i64)(in, slice_plan, out);
+            break;
+            TYPE_CASE(u8)(in, slice_plan, out);
+            break;
+            TYPE_CASE(u16)(in, slice_plan, out);
+            break;
+            TYPE_CASE(u32)(in, slice_plan, out);
+            break;
+            TYPE_CASE(u64)(in, slice_plan, out);
+            break;
+            TYPE_CASE(bf16)(in, slice_plan, out);
+            break;
+            TYPE_CASE(f32)(in, slice_plan, out);
+            break;
+            TYPE_CASE(f64)(in, slice_plan, out);
+            break;
+        default: rc = false; break;
+        }
+        return rc;
+    }
+
+    bool constant_fold_strided_slice(Node* strided_slice_node,
```

Comment:
Won't evaluate handle this?

Reply:
no. I referred to the `op::v0::ShapeOf::constant_fold`. 
Are you suggesting to put the folding into evaluate? Could you point me to the latest example?